### PR TITLE
BUG: Fix bug with logging and n_jobs>1

### PR DESCRIPTION
**Pull request by larsoner**
cc @hoechenberger I can replicate -- locally the added test passes with `n_jobs=1` and fails when `n_jobs=2`:
```
mne/utils/tests/test_logging.py:276: in test_verbose_threads
    assert out == want_levels
E   assert [20, 20, 20, 20, 20] == [30, 30, 30, 30, 30]
E     At index 0 diff: 20 != 30
E     Use -v to get more diff
```
So somehow we need to get the `logger.level` inside the `parallel_func`'ed function to persist. Not sure what it'll take but I'll try to look at some point

### CODE REVIEW

**Comment by larsoner** [comment]
@hoechenberger I can replicate your issue in MNE-BIDS-Pipeline on mne `main` with:
```
$ MNE_LOGGING_LEVEL=info pytest mne_bids_pipeline -k ds001810
=============================================================================== test session starts ================================================================================
platform linux -- Python 3.11.4, pytest-7.4.0, pluggy-1.0.0 -- /home/larsoner/python/virtualenvs/base/bin/python3
...
┌────────┬ preprocessing/_01_data_quality ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
│16:45:45│ ⏳️ sub-01 ses-anodalpost Initializing report HDF5 file
│16:45:45│ ⏳️ sub-01 ses-anodalpre Initializing report HDF5 file
Saving report to : /home/larsoner/mne_data/derivatives/mne-bids-pipeline/ds001810/sub-01/ses-anodalpre/eeg/sub-01_ses-anodalpre_task-attentionalblink_report.h5
Saving report to : /home/larsoner/mne_data/derivatives/mne-bids-pipeline/ds001810/sub-01/ses-anodalpost/eeg/sub-01_ses-anodalpost_task-attentionalblink_report.h5
Embedding : jquery-3.6.0.min.js
Embedding : bootstrap.bundle.min.js
Embedding : bootstrap.min.css
Embedding : bootstrap-table/bootstrap-table.min.js
Embedding : bootstrap-table/bootstrap-table.min.css
```
And on this PR I get the same thing, as I need to port the change over to MNE-BIDS-Pipeline. But making the change there I get:
```
$ MNE_LOGGING_LEVEL=info pytest mne_bids_pipeline -k ds001810
=============================================================================== test session starts ================================================================================
platform linux -- Python 3.11.4, pytest-7.4.0, pluggy-1.0.0 -- /home/larsoner/python/virtualenvs/base/bin/python3
...
┌────────┬ preprocessing/_01_data_quality ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
│16:50:18│ ⏳️ sub-01 ses-anodalpre Adding original raw data to report
│16:50:18│ ⏳️ sub-01 ses-cathodalpost Adding original raw data to report
│16:50:18│ ⏳️ sub-01 ses-anodalpost Adding original raw data to report
│16:50:18│ ⏳️ sub-01 ses-anodaltDCS Adding original raw data to report
...
```

**Review by hoechenberger — APPROVED** [review]
This looks good to me!

**Comment by hoechenberger** [comment]
Care to add a changelog entr?

File: doc/changes/devel.rst [hunk 1]
```diff
@@ -57,6 +57,7 @@ Bugs
 - Fix bug with :func:`mne.channels.read_ch_adjacency` (:gh:`11608` by :newcontrib:`Ivan Zubarev`)
 - Fix bugs with saving splits for :class:`~mne.Epochs` (:gh:`11876` by `Dmitrii Altukhov`_)
 - Fix bug with multi-plot 3D rendering where only one plot was updated (:gh:`11896` by `Eric Larson`_)
+- Fix bug where ``verbose`` level was not respected inside parallel jobs (:gh:`12154` by `Eric Larson`_)
```
Comment (larsoner) [inline doc/changes/devel.rst:60]: Change logged already @hoechenberger, could be you were looking before I pushed though

**Review by drammock — APPROVED** [review]

File: doc/changes/devel.rst (hunk 1 above)
Comment (hoechenberger) [reply doc/changes/devel.rst:60]: Thanks!

**Review by hoechenberger — APPROVED** [review]

**Pull request merged**